doc/rewriting-services.md

```diff
@@ -1,5 +1,50 @@
 # Tips for rewriting legacy services
 
+## Background
+
+The services are in the process of being rewritten to use our new service
+framework (#1584). Meanwhile, the legacy services extend from an abstract
+adapter called [LegacyService][] which provides a place to put the
+`camp.route()` invocation. The wrapper extends from [BaseService][], so it
+supports badge examples via `category`, `examples`, and `route`. Setting `route`
+also enables `create-service-tester.js` to infer a service's base path, reducing
+boilerplate for [creating the tester][creating a tester].
+
+Legacy services look like:
+
+```js
+camp.route(
+  /^\/example\/([^\/]+)\/([^\/]+)\.(svg|png|gif|jpg|json)$/,
+  cache(function(data, match, sendBadge, request) {
+    var first = match[1]
+    var second = match[2]
+    var format = match[3]
+    var badgeData = getBadgeData('X' + first + 'X', data)
+    badgeData.text[1] = second
+    badgeData.colorscheme = 'blue'
+    badgeData.colorB = '#008bb8'
+    sendBadge(format, badgeData)
+  })
+)
+```
+
+References:
+
+- [Old tutorial on legacy services][old tutorial]
+- Current documentation
```

Comment:
This is a really minor point and I might be over-thinking it slightly, but shall we put the link to the old docs further down the list? From the perspective of a newcomer, the first link in this document is to a tutorial that reflects bad practice. I can see why its a useful link, but it seems like a good idea to make the docs that reflect good practice the top thing in the list.